packages/otel/src/create-instrumented-graphql-client.ts

```diff
@@ -0,0 +1,12 @@
+import { CreateGraphQLClientArgs, createGraphQLClient } from "@saleor/apps-shared";
```

Comment:
Im not convinced, it leads to circular dependencies. 

I'd rather stick to Apps importing packages and avoid packages importing packages.

Best if we can export "factories" only in packages and app itself will compose these factories to create end-instances.

E.g. see how "logger" is used:
1. apps-logger exports logger
2. otel exports transport for logger
3. app imports both and assigns transport to logger